### PR TITLE
Removes numerous globals

### PR DESCRIPTION
**Pull request by johnSchnake**
Mainly globals for flag definitions replaced with variables
and making methods return closures around those variables rather
than accessing the globals.

There are other globals but this avoids one of the most common
types of bugs we've ran into and closes a long-standing issue
to fix these.

Fixes: 1052

Signed-off-by: John Schnake <jschnake@vmware.com>

### CODE REVIEW

File: cmd/sonobuoy/app/gen_config.go [hunk 1]
```diff
@@ -13,10 +13,11 @@ import (
 // NewCmdGenConfig creates the `config` command which will print out
 // the default sonobuoy config in a json format.
 func NewCmdGenConfig() *cobra.Command {
+	var f genFlags
 	var GenCommand = &cobra.Command{
 		Use:   "config",
 		Short: "Generates a sonobuoy config for input to sonobuoy gen or run.",
-		Run:   genConfigCobra,
+		Run:   genConfigCobra(f),
```
Comment (vladimirvivien) [inline cmd/sonobuoy/app/gen_config.go:20]: Other command values use a pointer param in their helper function for Run.

File: cmd/sonobuoy/app/logs.go [hunk 2]
```diff
@@ -32,58 +32,61 @@ const (
...
+		for {
+			n, err := logreader.Read(b)
+			if err != nil && err != io.EOF {
+				errlog.LogError(errors.Wrap(err, "error reading logs"))
+				os.Exit(1)
+			}
+			fmt.Fprint(logConfig.Out, string(b[:n]))
+			if err == io.EOF {
+				return
+			}
```
Comment (vladimirvivien) [inline cmd/sonobuoy/app/logs.go:89]: Consider `io.Copy(logConfig.out, logreader)` since `fmt.Fprintf` is not used for any sort of formatting.

**Review by vladimirvivien — COMMENTED** [review]
Looks great, left couple comments

File: cmd/sonobuoy/app/logs.go (hunk 2 above)
Comment (johnSchnake) [reply cmd/sonobuoy/app/logs.go:89]: Its just an `Fprint`, not `Fprintf` and `io.Copy` would need a reader where here its just some raw bytes.

**Comment by johnSchnake** [comment]
Making change based on input and merging once green. Then moving on to fix some issues that _slightly_ touch on some of this code

**Pull request merged**